src/App.jsx

```diff
@@ -14,45 +16,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const ALPHABET = 'Alphabet';
+const LENGTH = 'Length';
+
+function getPreparedGoods(goods, sortFiled, isReversed) {
+  const preparedGoods = [...goods];
+
+  if (sortFiled) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortFiled) {
+        case ALPHABET:
+          return good1.localeCompare(good2);
+
+        case LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortFiled, setSortFiled] = useState('');
+  const [isReversed, setReverse] = useState(false);
+
+  let visibleGoods = getPreparedGoods(goodsFromServer, sortFiled, isReversed);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const handleReverseClick = () => {
+    setReverse(prev => !prev);
+  };
+
+  function reset() {
+    visibleGoods = goodsFromServer;
```

Comment:
this line is redundant, 'cause getPreparedGoods function will return you the default array